test/utils/update/beneficial_owners_managing_officers_data_fetch.spec.ts

```diff
@@ -34,6 +41,43 @@ describe("util beneficial owners managing officers data fetch", () => {
     expect(appData.update?.bo_mo_data_fetched).toBe(true);
   });
 
+  test("Should not set BO data in appData if no Company PSCs returned", async () => {
+    appData = { "transaction_id": "123", "overseas_entity_id": "456" };
+    mockGetCompanyPscService.mockReturnValue({ "items": [] });
+    await retrieveBeneficialOwners(req, appData);
+
+    expect(appData.beneficial_owners_individual?.[0].first_name).toEqual(undefined);
+    expect(appData.beneficial_owners_corporate?.[0].name).toEqual(undefined);
+    expect(appData.beneficial_owners_government_or_public_authority?.[0].name).toEqual(undefined);
+  });
+
+  test("Should not set BO data in appData if no Company PSCs exist", async () => {
+    appData = { "transaction_id": "123", "overseas_entity_id": "456" };
+    mockGetCompanyPscService.mockReturnValue(null);
+    await retrieveBeneficialOwners(req, appData);
+
+    expect(appData.beneficial_owners_individual?.[0].first_name).toEqual(undefined);
+    expect(appData.beneficial_owners_corporate?.[0].name).toEqual(undefined);
+    expect(appData.beneficial_owners_government_or_public_authority?.[0].name).toEqual(undefined);
+  });
+
+  test("Ceased on test ", async () => {
+    appData = { "transaction_id": "123", "overseas_entity_id": "456" };
+    mockGetCompanyPscService.mockReturnValue( { "items": [ BO_INDIVIDUAL_MOCK_CEASED ] });
+    await retrieveBeneficialOwners(req, appData);
+
+    expect(appData.beneficial_owners_individual?.length).toEqual(undefined);
```

Comment:
Should this be `appData.update?.review_beneficial_owners_individual?.length`? That's if we want to check the data hasn't been set

Reply:
good spot - I missed that